src/java.base/share/classes/java/security/Provider.java

```diff
@@ -1106,16 +1069,15 @@ private ServiceKey(String type, String algorithm, boolean intern) {
             this.algorithm = intern ? algorithm.intern() : algorithm;
         }
         public int hashCode() {
-            return Objects.hash(type, algorithm);
+            return type.hashCode() ^ algorithm.hashCode();
```

Comment:
To avoid any suspicion we'd generate a worse hash here I've reverted this to inline what would be generated going through Objects.hash: `31*31 + type.hashCode()*31 + algorithm.hashCode()`